### PR TITLE
Cleaned up and fixed compiler warnings.

### PR DESCRIPTION
**Pull request by icarter09**
Added try/catch blocks and missing exceptions where needed. If anything looks wrong or needs to be tuned up, just let me know and I'll make the changes accordingly. Thanks.

### CODE REVIEW

File: src/slick-greeter.vala [hunk 1]
```diff
@@ -304,29 +312,61 @@ public class SlickGreeter
...
+    public void authenticate (string? userid = null)
+    {
+	try {
+		greeter.authenticate (userid);
+	}
+	catch(Error e) {
+		warning ("Unable to authenticate greeter for %s, %s", userid, e.message);
+	}
+    }
+/*
```
Comment (leigh123linux) [inline src/slick-greeter.vala]: Can't the commented code be removed?

Comment (icarter09) [reply src/slick-greeter.vala]: I'm an idiot and should have caught that. Removing it now.

**Comment by mtwebster** [comment]
Hi, can I squash these when I merge?  Or can you?  I think they can be in a single commit, since they're addressing the same issue over multiple files.

Thanks

**Comment by icarter09** [comment]
@mtwebster feel free to squash them. I'll be able to do it later tonight if need be.

**Pull request merged**